pkg/loadbalancer/reconcile/configmap.go

```diff
@@ -0,0 +1,186 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconcile
+
+import (
+	"context"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/open-telemetry/opentelemetry-operator/pkg/loadbalancer"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/naming"
+)
+
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+
+// ConfigMaps reconciles the config map(s) required for the instance in the current context.
+func ConfigMaps(ctx context.Context, params Params) error {
+	desired := []corev1.ConfigMap{}
+	cm, err := desiredConfigMap(ctx, params)
+	if err != nil {
+		return fmt.Errorf("failed to parse config: %v", err)
+	}
```

Comment:
I would move this into the block guarded by `checkMode()` since its results are only interesting/useful if a LB is required.